### ISSUE TITLE
Issues of magit-ediff-use-indirect-buffers

### ISSUE THREAD

**Issue by geza-herman** [magit/magit#4440]
I've found two issues of the new indirect-buffers feature:
 1. indirect-buffer isn't get deleted, if it is created as buffer-B (buffer is always killed with `(ediff-kill-buffer-carefully ediff-buffer-A)`)
 2. if the buffer doesn't exist before `ediff`, indirect-buffer is not created. I think that it makes sense to always create the indirect-buffer. The point is that two buffers should always exist: one, which is the original, which doesn't have ediff highlights, and another one, which is the indirect-buffer, which is passed to `ediff`. So, no matter how the original buffer is created (is already loaded, or magit loads for me), it should not have ediff's highlight.

**Comment by tarsius** [comment]
1. Fixed.
2. I disagree. IMO it makes sense to avoid messing up an existing buffer but less so to create a new buffer, avoid messing that up, and then create another new buffer.

**Comment by geza-herman** [comment]
1. Thank you!
2. I understand that my suggestion seems weird. Here's my use case: I execute `ediff` for whatever reason. Maybe I do some copies between buffer-A and B (or do some other edits) and then I'd like to validate my changes. On another tab, I execute a compile. If there's a problem with my changes, emacs will load the buffer. Which will have ediff's highlights, if the buffer wasn't loaded before I executed `ediff`. Or, here's another way to look at this problem: suppose that emacs supported window-local highlights. Then `ediff` could use this hypothetical feature, and the whole problem wouldn't exist. The current indirect-buffer feature is just a workaround, it basically simulates window-local highlights. Which means that indirect-buffer must always be created, the buffer that `ediff` uses must be unique.

**Comment by tarsius** [comment]
This makes sense.

But I have to admit that I regret having agreed to offer this feature and that by now intend to remove it again.

While I wouldn't use this feature I do not think it is useless, just not for me. But it is a hack and likely to cause issues in the future. Such as bugs and requests to further improve it. As I have said I do not believe macros should be used for this sort of thing but have done so because in this case the alternatives would be even worse. So far the macro is ugly but acceptable, but I just tried implementing what you requested and that made it so much worse.

More importantly though ediff already has a feature that should serve your needs: while ediff is active press `h` to cycle between different highlighting styles, including no highlighting. If I had realized that that exists I would never have agreed to implement the indirect buffer feature.

Sorry!
